### PR TITLE
mayhem integration

### PR DESCRIPTION
**Pull request by evanrichter**

### CODE REVIEW

File: sources/ghostdog/src/main.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ fn run() -> Result<()> {
 }
```
Comment (xansec) [inline sources/ghostdog/src/main.rs:50]: I presume you made this public to access for fuzzing, maybe we can if-gate this so to not modify the main functionality?

File: sources/parse-datetime/fuzz/fuzz_targets/parse.rs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#![no_main]
```
Comment (xansec) [inline sources/parse-datetime/fuzz/fuzz_targets/parse.rs:1]: Is this part of bottlerocket? If not, it should go in another repo.

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: sources/ghostdog/fuzz/fuzz_targets/img_scanning.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+use libfuzzer_sys::fuzz_target;
+
+fuzz_target!(|data: &[u8]| {
+    // Prevent known panic in gptman-1.0.0 with inputs smaller than a sector size
+    if data.len() < 512 {
+        return;
+    }
+
+    let mut data = std::io::Cursor::new(data);
+    let _ = ghostdog::find_device_type(&mut data);
```
Comment (xansec) [inline sources/ghostdog/fuzz/fuzz_targets/img_scanning.rs:11]: This doesn't seem to be part of bottlerocket?

**Review by xansec — CHANGES_REQUESTED** [review]
I also notice not all of these targets generate testcases.

File: sources/ghostdog/src/main.rs (hunk 1 above)
Comment (evanrichter) [reply sources/ghostdog/src/main.rs:50]: this doesn't functionally change anything about the public api of the crate. Ghostdog is a binary crate that just produces an executable. For fuzzing, cargo fuzz relies on importing a library to fuzz an API. That's why I created the _src/lib.rs_ file (and it is only active under conditional compilation!). In short, internal functions that become `pub` have no effect on executable programs, but it is necessary to plumb it through the library interface.

File: sources/parse-datetime/fuzz/fuzz_targets/parse.rs (hunk 2 above)
Comment (evanrichter) [reply sources/parse-datetime/fuzz/fuzz_targets/parse.rs:1]: this is not a vendored library as far as I can tell. "parse-datetime" is not on crates.io, and it seems like a small utility library for them. which is pretty common in rust, to break out standalone things into crates (it helps organization and compile times)

File: sources/ghostdog/fuzz/fuzz_targets/img_scanning.rs (hunk 3 above)
Comment (evanrichter) [reply sources/ghostdog/fuzz/fuzz_targets/img_scanning.rs:11]: I have no idea if ghostdog is packaged with the bottlerocket OS, but that doesn't really matter right? It's part of this repo (it is first party code, not vendored), and it reads external input from files.

**Review by xansec — APPROVED** [review]
Ok, I'm fine with these changes. LGTM!

**Pull request merged**